xbmc/windowing/windows/VideoSyncD3D.cpp

```diff
@@ -87,20 +122,29 @@ void CVideoSyncD3D::Run(CEvent& stopEvent)
     // save the timestamp of this vblank so we can calculate how many vblanks happened next time
     LastVBlankTime = Now;
 
-    if ((Now - m_lastUpdateTime) >= systemFrequency)
+    if (!m_factory->IsCurrent())
     {
+      CreateDXGIFactory1(IID_PPV_ARGS(m_factory.ReleaseAndGetAddressOf()));
+
       float fps = m_fps;
       if (fps != GetFps())
         break;
     }
 
-    // because we had a vblank, sleep until half the refreshrate period because i think WaitForVBlank block any rendering stuf
-    // without sleeping we have freeze rendering
+#if defined(TARGET_WINDOWS_DESKTOP)
+    // Half-vsync sleep is not needed for UWP or Win > 7 (no freeze with back to back WaitForVBlank())
+    // It's actually a problem for UWP because of the lack of high resolution timers so exclude.
+    // Keep for desktop Windows for now, there is no need to break compatibility with Windows 7 on purpose.
```

Comment:
Windows 7 is not longer supported on Omega then if "is not needed for UWP or Win > 7" this code block can be removed?

Reply:
Yes it can but will definitely break Windows 7 according to previous comments and other sources on the internet.

There is a difference between not supporting Win 7 and knowingly breaking it. I'm not sure if we're doing the former or the latter for v21.
It's not really in the way here (except for UWP with insufficient timer accuracy), but removing would probably allow a smaller fix. I'd have to run some tests.

Reply:
From https://github.com/xbmc/xbmc/pull/17255 minimal Windows version required is 8.1. 

Is not  even possible install v21 on Windows 7 (installer blocks it).